liquidctl/driver/smbus.py

```diff
@@ -0,0 +1,275 @@
+
+        @property
+        def description(self):
+            return self._try_sysfs_read('name')
+
+        @property
+        def parent_vendor(self):
+            return self._try_sysfs_read_hex('device/vendor')
+
+        @property
+        def parent_device(self):
+            return self._try_sysfs_read_hex('device/device')
+
+        @property
+        def parent_subsystem_vendor(self):
+            return self._try_sysfs_read_hex('device/subsystem_vendor')
+
+        @property
+        def parent_subsystem_device(self):
+            return self._try_sysfs_read_hex('device/subsystem_device')
+
+        @property
+        def parent_driver(self):
+            try:
+                return Path(os.readlink(self._i2c_dev.joinpath('device/driver'))).name
+            except FileNotFoundError:
+                return None
+
+        def __str__(self):
+            if self.description:
+                return f'{self.name}: {self.description}'
+            return self.name
+
+        def __repr__(self):
+            def hexid(maybe):
+                if maybe is not None:
+                    return f'{maybe:#06x}'
+                return 'None'
+
+            return f'{self.__class__.__name__}: name: {self.name!r}, ' \
+                   f'description: {self.description!r}, ' \
+                   f'parent_vendor: {hexid(self.parent_vendor)}, ' \
+                   f'parent_device: {hexid(self.parent_device)}, ' \
+                   f'parent_subsystem_vendor: {hexid(self.parent_subsystem_vendor)}, ' \
+                   f'parent_subsystem_device: {hexid(self.parent_subsystem_device)}, ' \
+                   f'parent_driver: {self.parent_driver!r}'
+
+        def _try_sysfs_read(self, *sub, default=None):
+            try:
+                return self._i2c_dev.joinpath(*sub).read_text().rstrip()
+            except FileNotFoundError:
+                return default
+
+        def _try_sysfs_read_hex(self, *sub, default=None):
+            try:
+                return int(self._i2c_dev.joinpath(*sub).read_text(), base=16)
+            except FileNotFoundError:
+                return default
+
+
+class SmbusDriver(BaseDriver):
+    """Base driver class for SMBus devices."""
+
+    @classmethod
+    def probe(cls, smbus, **kwargs):
+        raise NotImplementedError()
+
+    @classmethod
+    def find_supported_devices(cls, **kwargs):
+        """Find devices specifically compatible with this driver."""
+        if sys.platform != 'linux':
+            return []
+
+        devs = filter(lambda x: isinstance(x, cls), LinuxI2c.find_devices(**kwargs))
+        return list(devs)
+
+    def __init__(self, smbus, description, vendor_id=None, product_id=None,
+                 address=None, **kwargs):
+        assert vendor_id and product_id and address is not None
+
+        self._smbus = smbus
+        self._description = description
+        self._vendor_id = vendor_id
+        self._product_id = product_id
```

Comment:
Should a note be added somewhere that these id's are actually the subsystem vendor and the subsystem product to remove some confusion when people are trying to make other drivers later?

Reply:
Added.